src/io/index.js

```diff
@@ -0,0 +1,303 @@
+            };
+
+            temporaryFileReader.onload = () => {
+                resolve(temporaryFileReader.result);
+            };
+            temporaryFileReader.readAsDataURL(inputFile);
+        });
+    }
+
+    // Creates a file from an encoded data uri
+    async createFileFromEncodedDataURI(path) {
+        return new Promise((resolve, reject) => {
+            var dataURL = path.substring(path.indexOf('data:'));
+            var blob = this.dataURLtoBlob(dataURL);
+            var paths = path.substring(0, path.indexOf('data:') - 1);
+            var fileName = pth.basename(paths);
+            var filePath = pth.dirname(paths) + "/";
+
+            const handleUpload = async blob => {
+                try {
+                    const fileContents = await this.readUploadedFileAsArrayBuffer(
+                        blob
+                    );
+                    var buf = new Int8Array(fileContents);
+                    var file = { name: fileName, path: filePath, buffer: buf };
+                    const result = await this.createFileFromArrayBuffer(file);
+                    result.path = filePath;
+                    resolve(result);
+                } catch (e) {
+                    reject(e.message);
+                }
+            };
+            handleUpload(blob);
+        });
+    }
+
+    // Generates a Data URI for the specified file
+    async getFileDataURL(path) {
+        return new Promise((resolve, reject) => {
+            const handle = async () => {
+                try {
+                    const fileInfo = await this.getFileInfo(path);
+                    var contents = fileInfo.body.contents;
+                    var name = fileInfo.body.name;
+                    var file = new File([contents], name, {
+                        type: fileInfo.type,
+                      });
+                    const fileDataURI = await this.readUploadedFileAsDataURL(
+                        file
+                    );
+                    resolve({name: name, dataurl: fileDataURI});
+                } catch (e) {
+                    reject(e.message);
+                }
+            };
+            handle();
+        });
+    }
+
+    // Uploads a file into filesystem
+    async createFileFromArrayBuffer(file) {
+        return new Promise((resolve, reject) => {
+            var buf = buffer(file.buffer);
+            fs.writeFile(file.path + file.name, buf, function(err) {
+                if (err) reject({ success: false, err: err });
+                else resolve({ success: true });
+            });
+        });
+    }
+
+    // Uploads array of files into filesystem
+    async createFilesFromArrayBuffer(files) {
+        const promises = files.map(async file => {
+            file.buffer = Object.values(file.buffer);
+            const result = await this.createFileFromArrayBuffer(file);
+            return result;
+        });
+
+        const results = await Promise.all(promises).catch(function(err) {
```

Comment:
You should directly return the `Promise.all` and let the caller handle the `catch`.